core/app/models/spree/order_update_attributes.rb

```diff
@@ -0,0 +1,36 @@
+module Spree
+  class OrderUpdateAttributes
```

Comment:
I am also a bit lost on the purpose of this class. Although the apparent intention is to function as a converter of html parameters to the model fields, 3 such methods are outside of it in a controller helper.

If that was not the intention, but instead to do business logic on order & payments, then would you not be effectively taking logic out of the Checkout state machine and placing it here. Is that the intention?

Do you not think it will be better to separate the business logic from the persistence? Otherwise, you will be continuing on the Spree path of "every class/method is on its own" - let's parse and save everything just to be sure.

The name `OrderUpdateAttributes` is far from revealing. What else would one be updating on an order, if it is not the attributes? It handles payments, calls out to shipments .. I would rather suggest to call it `CheckoutProgressor`. 

But then again, I think that the full Checkout code logic has to be drawn out in with a UML sort of diagam, so that the message flow will become clear, and from the message flow - the needed classes will naturally take their place. 

Sandi Metz in her book [Practical Object-Oriented Design in Ruby](http://www.poodr.com/) talks about such message flow diagrams and about higher importance of messages over the classes.